File: app/src/main/java/com/shevelev/comics_viewer/activities/comics_creation/drag_lists/ListItemOnClickTouchListener.java
package com.shevelev.comics_viewer.activities.comics_creation.drag_lists;

import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;

import com.shevelev.comics_viewer.activities.comics_creation.ISortPagesActivityItemsEvents;
import com.shevelev.comics_viewer.common.structs.Area;
import com.shevelev.comics_viewer.common.structs.Point;
import com.shevelev.comics_viewer.common.structs.Size;
import com.shevelev.comics_viewer.common.helpers.ToastsHelper;

/**
 * Show hint after click list item
 */
public class ListItemOnClickTouchListener implements AdapterView.OnItemClickListener, AdapterView.OnTouchListener
{
        private final ISortPagesActivityItemsEvents events;

    private Point touchPos;         // Touch position in listView

    public ListItemOnClickTouchListener(ISortPagesActivityItemsEvents events)
    {
        this.events = events;
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id)
    {
        if(touchPos==null)
            return;

        Area area=getAreaByTag(view, "zoomIcon", 0, 0);
        if(area.isHit(touchPos))
            events.onZoomItem(position);
        else
        {
            area=getAreaByTag(view, "visibilityIcon", 0, 0);
            if(area.isHit(touchPos))
                events.onSetVisibilityItem(position);
            else
                ToastsHelper.Show(           // Show tag with item's name
                        ((ListItemDrag) (parent.getItemAtPosition(position))).getItemLongString(),
                        ToastsHelper.Position.Bottom);
        }
        touchPos=null;

    }

    @Override
    public boolean onTouch(View v, MotionEvent event)
    {
        if(event.getAction()==MotionEvent.ACTION_UP)
            touchPos=new Point((int)event.getX(), (int)event.getY());

        return false;
    }

    /**
     * Get control area with some tag
     * @param view
     * @param lookedTag
     * @param top
     * @param left
     * @return
     */
    private Area getAreaByTag(View view, String lookedTag, int top, int left)
    {
        left+=view.getLeft();
        top+=view.getTop();

        Object tag= view.getTag();
        if(tag!=null && tag.equals(lookedTag))
            return new Area(new Point(left, top), new Size(view.getWidth(), view.getHeight()));

        if(view instanceof ViewGroup)
        {
            ViewGroup viewGroup = (ViewGroup)view;
            for(int i=0; i<viewGroup.getChildCount(); i++)
            {
                Area area = getAreaByTag(viewGroup.getChildAt(i), lookedTag, top, left);
                if(area!=null)
                    return area;
            }
        }
        return null;
    }
}

File: app/src/main/java/com/shevelev/comics_viewer/activities/comics_creation/drag_lists/ListItemOnClickTouchListener.kt
package com.shevelev.comics_viewer.activities.comics_creation.drag_lists

import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import com.shevelev.comics_viewer.activities.comics_creation.ISortPagesActivityItemsEvents
import com.shevelev.comics_viewer.common.helpers.ToastsHelper
import com.shevelev.comics_viewer.common.structs.Area
import com.shevelev.comics_viewer.common.structs.Point
import com.shevelev.comics_viewer.common.structs.Size

/**
 * Show hint after click list item
 */
class ListItemOnClickTouchListener(private val events: ISortPagesActivityItemsEvents) : OnItemClickListener, OnTouchListener {
    private var touchPos : Point? = null        // Touch position in listView

    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        if (touchPos == null) {
            return
        }

        var area = getAreaByTag(view, "zoomIcon", 0, 0)
        if (area!!.isHit(touchPos)) events.onZoomItem(position) else {
            area = getAreaByTag(view, "visibilityIcon", 0, 0)
            if (area!!.isHit(touchPos)) events.onSetVisibilityItem(position) else ToastsHelper.Show( // Show tag with item's name
                (parent.getItemAtPosition(position) as ListItemDrag).itemLongString,
                ToastsHelper.Position.Bottom)
        }
        touchPos = null
    }

    override fun onTouch(v: View, event: MotionEvent): Boolean {
        if (event.action == MotionEvent.ACTION_UP) {
            touchPos = Point(event.x.toInt(), event.y.toInt())
        }
        return false
    }

    /**
     * Get control area with some tag
     */
    private fun getAreaByTag(view: View, lookedTag: String, top: Int, left: Int): Area? {
        val topInner = top + view.left
        val leftInner = left + view.left

        val tag = view.tag
        if (tag != null && tag == lookedTag) return Area(Point(leftInner, topInner), Size(view.width, view.height))
        if (view is ViewGroup) {
            for (i in 0 until view.childCount) {
                val area = getAreaByTag(view.getChildAt(i), lookedTag, topInner, leftInner)
                if (area != null) return area
            }
        }
        return null
    }
}